api-inference-community/docker_images/speechbrain/app/pipelines/audio_to_audio.py

```diff
@@ -1,32 +1,48 @@
-import json
+from enum import Enum
 from typing import List, Tuple
 
 import numpy as np
-import requests
 import torch
 from app.pipelines import Pipeline
+from huggingface_hub import HfApi
 from speechbrain.pretrained import SepformerSeparation, SpectralMaskEnhancement
 
 
-def get_info(model_id: str):
-    ENDPOINT = "https://huggingface.co/api/models/"
-    response = requests.get(f"{ENDPOINT}{model_id}")
-    if response.status_code != 200:
-        raise Exception("Cannot infer the code properly, please set some tags")
-    model_info = json.loads(response.content.decode("utf-8"))
-    tags = [
-        tag.lower().replace(" ", "-").replace("_", "-") for tag in model_info["tags"]
-    ]
-    return tags
+class ModelType(Enum):
+    AUDIO_SOURCE_SEPARATION = 1
+    SPEECH_ENHANCEMENT = 2
+
+
+def interface_to_type(interface_str):
+    if interface_str == "SepformerSeparation":
+        return ModelType.AUDIO_SOURCE_SEPARATION
+    elif interface_str == "SpectralMaskEnhancement":
+        return ModelType.SPEECH_ENHANCEMENT
+    else:
+        raise ValueError(f"Invalid interface: {interface_str} for Audio to Audio.")
+
+
+def get_type(model_id):
+    info = HfApi().model_info(repo_id=model_id)
+    if info.config:
+        if "speechbrain" in info.config:
+            interface_str = info.config["speechbrain"].get(
+                "interface", "SepformerSeparation"
+            )
+        else:
+            interface_str = "SepformerSeparation"
+    else:
+        interface_str = "SepformerSeparation"
+    return interface_to_type(interface_str)
```

Comment:
Taste comments (feel free to ignore):

- I would use an enum covering all cases for all pipelines and have code `interface_to_type` be very generic.
- I would use same names as the strings in the configs:
  1. Would improve readability/reduce confusion
  2. Would enable `ModelType[interface_str]` code, which is probably easier to maintain than a series of if/else. (`ModelType[interface_str.upper()]` if we want to keep uppercase in the enum
- I would use STR values instead of integers in the enum. Less confusing when errors pop up, and shouldn't have any disadvantage over ints.

Other NIT, it's a bit longer but IMO shows intention a bit better.:

```python
def get_type(model_id):
    info = HfApi().model_info(repo_id=model_id)
    DEFAULT = "SepformerSeparation"
    if info.config:
        interface_str = info.config.get("speechbrain", {}).get("interface", DEFAULT)
    return interface_to_type(interface_str)
```

Reply:
Good feedback, thanks a lot

* I changed the enum to cover all cases
* I like that we can do `ModelType[interface_str]` although it's a bit less readable `ENCODERDECODERASR` now vs `ENCODER_DECODER_ASR` before
* I changed the value to be strings, which is also useful for error handling
* I decided to remove having a default interface entirely instead of your last suggestion. I feel it's better to not do any magic from our side. Having a default interface will likely lead to unintended consequences and unintuitive behavior for users. If there's no config or no interface specified in the config, let's just give an error.